python/tank/descriptor/io_descriptor/git_branch.py

```diff
@@ -134,37 +110,50 @@ def download_local(self):
         """
         Retrieves this version to local repo.
         Will exit early if app already exists local.
+
+        This will connect to remote git repositories.
+        Depending on how git is configured, https repositories
+        requiring credentials may result in a shell opening up
+        requesting username and password.
+
+        The git repo will be cloned into the local cache and
+        will then be adjusted to point at the relevant commit.
         """
         if self.exists_local():
             # nothing to do!
             return
 
         # cache into the primary location
         target = self._get_cache_paths()[0]
-        self._clone_into(target)
 
-    def copy(self, target_path, connected=False):
-        """
-        Copy the contents of the descriptor to an external location
-
-        :param target_path: target path to copy the descriptor to.
-        :param connected: For descriptor types that supports it, attempt
-                          to create a 'connected' copy that has a relationship
-                          with the descriptor. This is typically useful for SCMs
-                          such as git, where rather than copying the content in
-                          its raw form, you clone the repository, thereby creating
-                          a setup where changes can be made and pushed back to the
-                          connected server side repository.
-        """
-        if connected:
-            self._clone_into(target_path)
-        else:
-            super(IODescriptorGitBranch, self).copy(target_path, connected)
+        try:
+            # clone the repo, switch to the given branch
+            # then reset to the given commit
+            commands = [
+                "checkout -q \"%s\"" % self._branch,
+                "reset --hard -q \"%s\"" % self._version
+            ]
+            self._clone_then_execute_git_command(target, commands)
+
+        except Exception, e:
+            raise TankDescriptorError(
+                "Could not locally cache %s, branch %s, "
```

Comment:
minor: since the method is called `download_local`, the error might better as `Could not download %s,...`